tests/unit/schema/test_api.py

```diff
@@ -2602,3 +2604,215 @@ def test_has_iam_mixin():
+            auto_populated_fields=["squid", "mollusc"],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+        "google.example.v1beta1.SomeExample.Example2": client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example2",
+            auto_populated_fields=[],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+        "google.example.v1beta1.SomeExample.Example3": client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example3",
+            auto_populated_fields=[],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+        "google.example.v1beta1.SomeExample.Example4": client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example4",
+            auto_populated_fields=[],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+        "google.example.v1beta1.SomeExample.Example5": client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example5",
+            auto_populated_fields=[],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+    }
+
+    # Test that ValueError is raised when the rpc in publishing.method_settings.selector is not a unary one
+    opts = Options(
+        service_yaml_config={
+            "apis": [
+                {"name": "google.example.v1beta1.SomeExample.Example1"},
+            ],
+            "publishing": {
+                "method_settings": [
+                    {
+                        "selector": "google.example.v1beta1.DoesNotExist.Example1",
+                        "auto_populated_fields": [
+                            "squid",
+                            "mollusc",
+                        ],
+                    },
+                    {
+                        "selector": "google.example.v1beta1.SomeExample.Example1",
+                        "auto_populated_fields": [
+                            "doesnotexist",
+                            "mollusc",
+                            "octopus",
+                        ]
+                    },
+                    {
+                        "selector": "google.example.v1beta1.SomeExample.Example2",
+                        "auto_populated_fields": [
+                            "squid",
+                            "mollusc",
+                        ],
+                    },
+                    {
+                        "selector": "google.example.v1beta1.SomeExample.Example3",
+                        "auto_populated_fields": [
+                            "squid",
+                            "mollusc",
+                        ],
+                    }
+                ]
+            },
+        }
+    )
+    api_schema = api.API.build(fd, "google.example.v1beta1", opts=opts)
+
+    exception_message = """Fields cannot be automatically populated in the top level \
```

Comment:
We shouldn't compare exact text of error messages. We don't want users to rely on that and we shouldn't either. I suggest checking the custom error type (if you go with my other suggestion) and then checking that the error message contains the relevant strings (eg selector strings defined in this test).

Reply:
Fixed in 5a2d33b8b160a246945e3752e145878581a351b1